src/libcore/slice.rs

```diff
@@ -666,6 +666,60 @@ macro_rules! iterator {
                 let exact = diff / (if size == 0 {1} else {size});
                 (exact, Some(exact))
             }
+
+            #[inline]
+            fn count(self) -> usize {
```

Comment:
This whole comment thread says to me that we need a specific `consume` method for just running through an iterator.